ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/LedgerDB/OrphanArbitrary.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE TypeApplications #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+module Test.Ouroboros.Storage.LedgerDB.OrphanArbitrary where
+
+import           Test.QuickCheck
+import           Data.Time.Clock (DiffTime, secondsToDiffTime, diffTimeToPicoseconds)
+import           Ouroboros.Consensus.Config.SecurityParam (SecurityParam(..))
+import           Ouroboros.Consensus.Storage.LedgerDB.DiskPolicy (RequestedSnapshotInterval(..))
+
+{-------------------------------------------------------------------------------
+  Orphan Arbitrary instances
+-------------------------------------------------------------------------------}
+
+instance Arbitrary SecurityParam where
+  arbitrary = SecurityParam <$> choose (0, 6)
+  shrink (SecurityParam k) = SecurityParam <$> shrink k
+
+instance Arbitrary RequestedSnapshotInterval where
+  arbitrary =
+    oneof [ requestedSnapshotInterval
+          , defaultSnapshotInterval
+          ]
+  shrink DefaultSnapshotInterval = []
+  shrink (RequestedSnapshotInterval v) = RequestedSnapshotInterval <$> shrink v
+
+requestedSnapshotInterval :: Gen RequestedSnapshotInterval
+requestedSnapshotInterval =
+  RequestedSnapshotInterval <$> arbitrary
+
+defaultSnapshotInterval :: Gen RequestedSnapshotInterval
+defaultSnapshotInterval = pure DefaultSnapshotInterval
+
+instance Arbitrary DiffTime where
+  arbitrary = fromIntegral @Int <$> arbitrary
+  shrink dt = secondsToDiffTime <$> shrink (diffTimeToPicoseconds dt)
```

Comment:
Hmm. Without double-checking the definitions, just reading the names here sounds like this `shrink` will be increasing the value by a factor of `Pico`. Can you check? If the names are merely misleading, would you add a comment assuring anyone else who's confused?

Reply:
```
λ show <$> getCurrentTime 
"2021-03-23 13:25:27.497907 UTC"

λ show . utctDayTime <$> getCurrentTime 
"48331.201854s"

λ diffTimeToPicoseconds . utctDayTime <$> getCurrentTime 
48338194408000000

λ shrinkIntegral . diffTimeToPicoseconds . utctDayTime <$> getCurrentTime 
[0,24182675636500000,36274013454750000,42319682363875000,45342516818437500,46853934045718750,47609642659359375,47987496966179688,48176424119589844,48270887696294922,48318119484647461,48341735378823731,48353543325911866,48359447299455933,48362399286227967,48363875279613984,48364613276306992,48364982274653496,48365166773826748,48365259023413374,48365305148206687,48365328210603344,48365339741801672,48365345507400836,48365348390200418,48365349831600209,48365350552300105,48365350912650053,48365351092825027,48365351182912514,48365351227956257,48365351250478129,48365351261739065,48365351267369533,48365351270184767,48365351271592384,48365351272296192,48365351272648096,48365351272824048,48365351272912024,48365351272956012,48365351272978006,48365351272989003,48365351272994502,48365351272997251,48365351272998626,48365351272999313,48365351272999657,48365351272999829,48365351272999915,48365351272999958,48365351272999979,48365351272999990,48365351272999995,48365351272999998,48365351272999999]
```